Navigation/include/ActsLUXEPipeline/LUXEMeasurementsCreator.hpp

```diff
@@ -0,0 +1,246 @@
+            // Set options for propagator
+            Acts::PropagatorOptions<Actions, Aborters> options(ctx.geoContext, ctx.magFieldContext);
+
+            auto& creator = options.actionList.get<LUXENavigator::MeasurementsCreatorAction>();
+            creator.sourceId = ctx.eventNumber;
+
+            // Launch and collect the measurements
+            auto result = propagator.propagate(trackParameters, options).value();
+            return std::move(result.template get<LUXEDataContainer::SimMeasurements>());
+        }
+
+        /// @brief The execute method
+        ProcessCode execute(const AlgorithmContext& ctx) const override {
+            using namespace Acts::UnitLiterals;
+
+            // Create IP covariance matrix from
+            // reasonable standard deviations
+            Acts::BoundVector ipStdDev;
+            ipStdDev[Acts::eBoundLoc0] = 100_um;
+            ipStdDev[Acts::eBoundLoc1] = 100_um;
+            ipStdDev[Acts::eBoundTime] = 25_ns;
+            ipStdDev[Acts::eBoundPhi] = 2_degree;
+            ipStdDev[Acts::eBoundTheta] = 2_degree;
+            ipStdDev[Acts::eBoundQOverP] = 1 / 100_GeV;
+            Acts::BoundSquareMatrix ipCov =
+                    ipStdDev.cwiseProduct(ipStdDev).asDiagonal();
+
+            // Create the measurements
+            std::random_device rd;
+            std::mt19937 gen(rd());
+            std::normal_distribution<> pDisP(0.002,0.0018);
+            std::normal_distribution<> pDisM(-0.002,0.0018);
+            std::gamma_distribution<double> pzDis(3, 1.2);
+
+            Acts::ActsScalar m_e = 0.000511;
+            LUXEDataContainer::SimMeasurements results;
+            Acts::ActsScalar px = (pDisP(gen)+pDisM(gen))/2;
+            Acts::ActsScalar pz = (pDisP(gen)+pDisM(gen))/2;
+            Acts::ActsScalar py = pzDis(gen)+1;
+            Acts::ActsScalar p = std::sqrt(std::pow(px,2)+std::pow(py,2)+std::pow(pz,2));
+            Acts::ActsScalar E = std::hypot(p,m_e);
+            Acts::ActsScalar theta = std::acos(pz / p);
+            Acts::ActsScalar phi = std::atan2(py, px);
+            Acts::Vector4 mPos4{0,0,0,0};
+
+            Acts::Experimental::DetectorNavigator::Config cfg{&(*m_cfg.detector)};
+            cfg.resolvePassive = false;
+            cfg.resolveMaterial = true;
+            cfg.resolveSensitive = true;
+            Acts::Experimental::DetectorNavigator navigator(
+                    cfg, Acts::getDefaultLogger("Detector Navigation", Acts::Logging::VERBOSE));
+            Acts::EigenStepper<> stepper(std::move(m_cfg.BFieldPtr));
+
+            auto propagator = Acts::Propagator<decltype(stepper), Acts::Experimental::DetectorNavigator>(
+                    std::move(stepper), std::move(navigator));
+            results = createMeasurements(
+                    propagator, ctx,
+                    Acts::CurvilinearTrackParameters(mPos4, phi*180/M_PI*1_degree, theta*180/M_PI*1_degree,
```

Comment:
In `UnitLiterals` there's a "_deg" multiplier. Also, can't you take the rotation from the `GeometryOptions`?